pkg/openstack/horizon.go

```diff
@@ -73,6 +82,9 @@ func ReconcileHorizon(ctx context.Context, instance *corev1beta1.OpenStackContro
 		return ctrl.Result{}, err
 	}
 
+	// Set HSTS Headers for Horizon Route
+	instance.Spec.Horizon.APIOverride.Route.Annotations = HSTSAnnotation
```

Comment:
* I suggest to make this additive to any user passed annotation in instance.Spec.Horizon.APIOverride.Route.Annotations. You can user utils.MergeMaps to add your annotation to the existing ones.

* I'm a bit reluctant to modify instance.Spec directly as this is misleading. The controller should never modify its own Spec and I think to avoid that we explicitly ignore spec changes when persisting the instance at the end of the reconcile.

  To make it less confusing you could take a deep copy instance.Spec.Horizon.APIOverride, modify the copy and then pass that copy to EnsureEndpointConfig.

  Alternatively you can modify the result of EnsureEndpointConfig to add your annotations in `endpointDetails.EndpointDetails[service.EndpointPublic].Route.OverrideSpec.Annotations`

Reply:
True, we don't need to actually modify the spec. We can just create a copy of the user provided values and add our own. I've pushed a new revision using that method.

Reply:
could you also merge the new annotations with any existing one coming from the user?

Reply:
It wouldn't need to in this case right? It's just a `map[string]string`, so in the latest revision:
```
	// Set HSTS Headers for Horizon Route
	apiOverrides := instance.Spec.Horizon.APIOverride
	apiOverrides.Route.Annotations[HorizonHSTSHeaderAnnotation] = HorizonHSTSHeaderAnnotationValue

```

We copy any of the user provided values when we duplicate the struct to our `apiOverrides` variable, then we just insert our annotation in addition to whatever already exists in the map.

Maybe the user provided something and the map looks like:
```
map[string]string{
  "this-blah-named-annotation": "this-blah-value",
  "more-blah-annotations": "more-blah-values",
  }
```

Then I come along and make a full copy of that and just add my own in, so now it looks like:
```
map[string]string{
  "this-blah-named-annotation": "this-blah-value",
  "more-blah-annotations": "more-blah-values",
  "haproxy.router.openshift.io/hsts_header": "max-age=31536000;includeSubDomains;preload",
  }
```

Or are you suggesting that the user might also provide a annotation for `"haproxy.router.openshift.io/hsts_header"` and I should merge their provided value and ours?

Maybe if that's the case, then I should just do "if this key does not exists, add annotation."

Reply:
Latest revision looks like this:
```
	// Set HSTS Headers for Horizon Route
	//
	apiOverrides := instance.Spec.Horizon.APIOverride

	if _, ok := apiOverrides.Route.Annotations[HorizonHSTSHeaderAnnotation]; !ok {
		apiOverrides.Route.Annotations[HorizonHSTSHeaderAnnotation] = HorizonHSTSHeaderAnnotationValue
	}

```

Reply:
ahh you are right, I looked at the wrong version. You are clearly adding to it not overriding it. Thanks